### ISSUE TITLE
The ability to detect os family. 

### ISSUE THREAD

**Issue by lowks** [aelsabbahy/goss#154]
I am very impressed with `goss` and want to move my `serverspec` tests to `goss`. One thing that I can't find how to do in `goss` is detecting os family. For example: 

```
if os[:family] == 'redhat'
    packages = ["git", "unzip", "epel-release", "nginx"]
elsif ['debian', 'ubuntu'].include?(os[:family])
    packages = ["git", "unzip", "uwsgi", "nginx"]
end
```

Is this feature already in `goss` or can it be added ?

**Comment by aelsabbahy** [comment]
So, I was personally thinking that this feature (the ability to probe system for "facts") should be out of scope for `goss`. That said, I do think `goss` should have some sort of support for variables. see: https://github.com/aelsabbahy/goss/issues/136#issuecomment-242773611

Do you think the approach posted in that comment would sufficiently cover your usecase?

**Comment by lowks** [comment]
yeah that looks okay to me.

**Comment by lowks** [comment]
Any updates on this ?

**Comment by aelsabbahy** [comment]
No, unfortunately I haven't had time to put into goss recently.

**Comment by aelsabbahy** [comment]
Released in [v0.3.0](https://github.com/aelsabbahy/goss/releases/tag/v0.3.0)
